// vscode-languageserver/lib/common/proposed.notebooks.js
/* --------------------------------------------------------------------------------------------
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 * ------------------------------------------------------------------------------------------ */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebooksFeature = void 0;
const vscode_languageserver_protocol_1 = require("vscode-languageserver-protocol");
const NotebooksFeature = (Base) => {
    return class extends Base {
        get synchronization() {
            return {
                onDidOpenNotebookDocument: (handler) => {
                    this.connection.onNotification(vscode_languageserver_protocol_1.Proposed.DidOpenNotebookDocumentNotification.type, (params) => {
                        handler(params);
                    });
                },
                onDidChangeNotebookDocument: (handler) => {
                    this.connection.onNotification(vscode_languageserver_protocol_1.Proposed.DidChangeNotebookDocumentNotification.type, (params) => {
                        handler(params);
                    });
                },
                onDidCloseNotebookDocument: (handler) => {
                    this.connection.onNotification(vscode_languageserver_protocol_1.Proposed.DidCloseNotebookDocumentNotification.type, (params) => {
                        handler(params);
                    });
                }
            };
        }
    };
};
exports.NotebooksFeature = NotebooksFeature;
